source/FreeRTOS_DNS_Cache.c

```diff
@@ -284,8 +348,14 @@
                 xDNSCache[ uxIndex ].ucCurrentIPAddress++;
             #endif /* if ( ipconfigDNS_CACHE_ADDRESSES_PER_ENTRY > 1 ) */
 
-            *pulIP = xDNSCache[ uxIndex ].ulIPAddresses[ ulIPAddressIndex ];
+            ( void ) memcpy( pxIP, &( xDNSCache[ index ].xAddresses[ ulIPAddressIndex ] ), sizeof( *pxIP ) );
```

Comment:
`index` is not defined
~~~diff
-    xDNSCache[ index ]
+    xDNSCache[ uxIndex ]
~~~